### PR TITLE
Add CBMC proof-running GitHub Action

### PR DESCRIPTION
**Pull request by karkhaz**
This commit adds a GitHub Action that runs the CBMC proofs upon every push and pull request. This is intended to replace the current CBMC CI.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/proof_ci.yaml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+# SPDX-License-Identifier: MIT-0
+# CBMC starter kit 2.9
+name: Run CBMC proofs
+on:
+  push:
+    branches-ignore:
+      - gh-pages
+  pull_request:
+    branches-ignore:
+      - gh-pages
```
Comment (graebm) [inline .github/workflows/proof_ci.yaml:11]: trivial: to my knowledge, the common-runtime team doesn't have a `gh-pages` branch in any of our repos
but if you just want this here for consistency with how it's set up in other repos, that's fine too

File: .github/workflows/proof_ci.yaml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+# If you need to use different versions for tools like CBMC, modify this file:
+# .github/workflows/proof_ci_resources/config.yaml
+#
+# If you want the CI to use a different GitHub-hosted runner (which must still
+# be running Ubuntu 20.04), modify the value of this key:
+# jobs.run_cbmc_proofs.runs-on
+
+jobs:
+  run_cbmc_proofs:
+    runs-on: cbmc_ubuntu-latest_64-core
```
Comment (graebm) [inline .github/workflows/proof_ci.yaml:25]: just curious, where does this customer runner `cbmc_ubuntu-latest_64-core` come from?

File: .github/workflows/proof_ci.yaml [hunk 3]
```diff
@@ -0,0 +1,188 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+# CBMC starter kit 2.9
+name: Run CBMC proofs
```
Comment (graebm) [inline .github/workflows/proof_ci.yaml:4]: Should I remove the old "CBMC Proofs" Codebuild jobs once this is merged?

**Review by graebm — APPROVED** [review]

File: .github/workflows/proof_ci.yaml (hunk 1 above)
Comment (karkhaz) [reply .github/workflows/proof_ci.yaml:11]: We do have a [central definition](https://github.com/model-checking/cbmc-starter-kit/blob/master/src/cbmc_starter_kit/template-for-ci-workflow/proof_ci.yaml) of this file that would be copied into this repository whenever we update the starter kit, so I suggest we keep this here for idempocy

File: .github/workflows/proof_ci.yaml (hunk 2 above)
Comment (karkhaz) [reply .github/workflows/proof_ci.yaml:25]: This is a [Large Runner](https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners) pool that we have access to. This pool is shared with all other CBMC-using repositories in the awslabs GitHub organisation and has a certain concurrency limit. I'll ping you offline with more details.

File: .github/workflows/proof_ci.yaml (hunk 3 above)
Comment (karkhaz) [reply .github/workflows/proof_ci.yaml:4]: I suggest we keep them running in parallel for a little while to bake and ensure that the new one works smoothly. I'll be working on migrating the data from the old accounts. Once that's done, I can ping you to set the new check as "Required" and remove the old webhook.

Comment (karkhaz) [reply .github/workflows/proof_ci.yaml:4]: To clarify, by "a little while" I expect ~3 weeks or so to do the data transfer.

File: .github/workflows/proof_ci_resources/config.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+cadical-tag: latest
+cbmc-version: latest
+cbmc-viewer-version: latest
+kissat-tag: latest
+litani-version: latest
+proofs-dir: verification/cbmc/proofs
+run-cbmc-proofs-command: ./run-cbmc-proofs.py
```
Comment (feliperodri) [inline .github/workflows/proof_ci_resources/config.yaml:7]: starter-kit is not part of this list, so I assume we will continue updating it as we always do: submitting a PR after applying the script the update it. Is that correct?

**Review by feliperodri — APPROVED** [review]
Thank you @karkhaz!

File: .github/workflows/proof_ci_resources/config.yaml (hunk 4 above)
Comment (karkhaz) [reply .github/workflows/proof_ci_resources/config.yaml:7]: This file is considered part of the starter kit, it's in the same repo :) so whenever we update the starter kit, the relevant files will change anyway

**Pull request merged**